common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerTokenProvider.java

```diff
@@ -0,0 +1,36 @@
+package com.microsoft.identity.common.internal.broker;
+
+import android.app.Activity;
+import android.content.Intent;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.exception.ErrorStrings;
+import com.microsoft.identity.common.internal.logging.Logger;
+
+
+public class BrokerTokenProvider {
+
+    private static final String TAG = BrokerTokenProvider.class.getSimpleName();
+
+    /**
+     * Acquire token interactively, will prompt user if possible.
+     * See {@link MicrosoftAuthServiceHandler#getIntentForInteractiveRequest(final Context context)} for details.
+     */
+    public void acquireTokenInteractively(final Activity activity)
+            throws ClientException {
+        final String methodName = ":acquireTokenInteractively";
+
+        if (activity == null) {
+            throw new ClientException(ErrorStrings.EMPTY_ANDROID_CONTEXT);
+        }
+
+        final Intent brokerIntent = MicrosoftAuthServiceHandler.getInstance().getIntentForInteractiveRequest(activity);
+        activity.startActivityForResult(brokerIntent, AuthenticationConstants.UIRequest.BROWSER_FLOW);
```

Comment:
is MSAL using AuthenticationConstants.UIRequest.BROWSER_FLOW?

Reply:
That's a constant int value representing the request code. As long as MSAL use this constant, it will work. In MSAL's activity's onActivityResult method, there should be a case to handle this constant.

Reply:
I agree, but how are you keeping track of the change on the MSAL side, while we wait for the MSAL change, can you atleast add a TODO here and in the MSAL code to handle this? I am also curious how this will interface with system webview.